qiita_db/processing_job.py

```diff
@@ -219,6 +286,95 @@ def status(self, value):
             qdb.sql_connection.TRN.add(sql, [new_status, self.id])
             qdb.sql_connection.TRN.execute()
 
+    def _generate_cmd(self):
+        """Generates the command to submit the job
+
+        Returns
+        -------
+        str
+            The command to use to submit the job
+        """
+        job_dir = join(qdb.util.get_work_base_dir(), self.id)
+        software = self.command.software
+        plugin_start_script = software.start_script
+        plugin_env_script = software.environment_script
+        cmd = '%s "%s" "%s" "%s" "%s" "%s"' % (
+            qiita_config.plugin_launcher, plugin_env_script,
+            plugin_start_script, qiita_config.base_url, self.id, job_dir)
+        return cmd
+
+    def submit(self):
+        """Submits the job to execution
+
+        Raises
+        ------
+        QiitaDBOperationNotPermittedError
+            If the job is not in 'waiting' or 'in_construction' status
+        """
+        if self.status not in {'in_construction', 'waiting'}:
+            raise qdb.exceptions.QiitaDBOperationNotPermittedError(
+                "Can't submit job, not in 'in_construction' or "
```

Comment:
For debugging purposes with this and the other exceptions, would it make sense to what the status is in the error message?